pennylane/measurements/measurements.py

```diff
@@ -185,7 +185,6 @@ def numeric_type(self):
             + f"return_type {self.return_type}."
         )
 
-    @functools.lru_cache()
     def shape(self, device=None):
```

Comment:
Is it necessary to keep the function the only thing that it does is to return an error?

Reply:
This way we have a nicer error message. If not we would obtain:
```
'MeasurementProcess' object has not attribute 'shape'
```